### ISSUE TITLE
onBlur callback doesn't get called when value is null

### ISSUE THREAD

**Issue by sceutre** [YouCanBookMe/react-datetime#251]
### I'm Submitting a ...

```
[x] Bug report
[ ] Feature request
[ ] Support request
```

### Steps to Reproduce

1. Use <ReactDatetime /> with an onBlur callback and value = null and closeOnSelect = true
2. Click in and select a date
3. Calendar closes but onBlur does not get called
4. Clicking elsewhere on the page also does not invoke onBlur

### Expected Results

onBlur callback should have been called when calendar closed

### Actual Results

onBlur callback was not called

### Minimal Reproduction of the Problem

```
// apologies for typescript format ...

export class DateTest extends React.Component<{}, { time?: number, key?: number }> {
   render() {
      let state = this.state || {};
      return <ReactDatetime
            key={state.key || 1}
            onChange={this.onChange}
            onBlur={this.onBlur}
            dateFormat={true}
            timeFormat={false}
            closeOnSelect={true}
            inputProps={{placeholder: "Select a date..."}}
            value={state.time == null ? null : new Date(state.time)}
         />; 
   }

   onChange = (val:any) => {
      if (typeof val == "string") {
         if (val == "") {
            this.setState({time: null, key: (this.state.key || 1) + 1});
         }
      } else {
         this.setState({ time: val.valueOf() });
      }
   };

   onBlur = (val:any) => {
      console.log("blur called", val, new Date());
      // this.onSave(val)
   };
}
```

### Other Information (e.g. stacktraces, related issues, suggestions how to fix)
 
If you're wondering why anyone would want to set the value to null, the example above shows our use case.  We wish to start with the placeholder text shown on the input field.

Additionally transitioning back to null doesn't really work either.  This seems like more of a react issue.  Anyway as you see in the example we sidestep that issue by just incrementing a key.

Taking a quick look at the code, it seems the issue lies in DateTime.js > updateSelectedDate.  Here is a snippet:
```
		if ( !this.props.value ){
			this.setState({
				selectedDate: date,
				viewDate: date.clone().startOf('month'),
				inputValue: date.format( this.state.inputFormat ),
				open: !(this.props.closeOnSelect && close )
			});
		} else {
			if (this.props.closeOnSelect && close) {
				this.closeCalendar();
			}
		}
```

The first branch in the if statement is what it getting called (since value is falsey) and it does not do an onBlur callback as it closes the calendar.  The else branch calls this.closeCalendar() which does do an onBlur callback.

**Comment by simeg** [comment]
Hi @sceutre, thanks for the detailed issue. I confirmed that this indeed is a bug. Great job investigating it, I will look into this when I find the time. Though it kinda looks like you have it worked out, so you are more than happy to submit a PR for the bug.

**Comment by simeg** [comment]
Hi @sceutre, this has been fixed in 794700cc3c0378368320255e978ee3b249a33a48 and will be included in the next release of the component. Will close this for now, if you're still having problems just let me know.
